From 25028a656faa061b0bbaa8b9a8a90ce755810a4b Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Wed, 30 Mar 2016 13:30:10 +0300
Subject: [PATCH 0133/1345] clk: ap806: Mark AP fixed clocks as root clocks

commit  54495e3a0f9384e8103e5335ba665d678225d826 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This enables other child clocks to use this clock as their source.

Change-Id: I4811fda8fa036c0a3c5ba9c64919051eac8da1cc
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28666
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/mvebu/ap806-system-controller.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index 8155bac..6fa7fa0 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -100,7 +100,7 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 	of_property_read_string_index(np, "clock-output-names",
 				      0, &name);
 	ap806_clks[0] = clk_register_fixed_rate(&pdev->dev, name, NULL,
-						0, cpuclk_freq);
+						CLK_IS_ROOT, cpuclk_freq);
 	if (IS_ERR(ap806_clks[0])) {
 		ret = PTR_ERR(ap806_clks[0]);
 		goto fail0;
@@ -108,7 +108,7 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 
 	of_property_read_string_index(np, "clock-output-names",
 				      1, &name);
-	ap806_clks[1] = clk_register_fixed_rate(&pdev->dev, name, NULL, 0,
+	ap806_clks[1] = clk_register_fixed_rate(&pdev->dev, name, NULL, CLK_IS_ROOT,
 						cpuclk_freq);
 	if (IS_ERR(ap806_clks[1])) {
 		ret = PTR_ERR(ap806_clks[1]);
@@ -118,7 +118,7 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 	/* Fixed clock is always 1200 Mhz */
 	of_property_read_string_index(np, "clock-output-names",
 				      2, &fixedclk_name);
-	ap806_clks[2] = clk_register_fixed_rate(&pdev->dev, fixedclk_name, NULL,
+	ap806_clks[2] = clk_register_fixed_rate(&pdev->dev, fixedclk_name, NULL, CLK_IS_ROOT,
 						0, 1200 * 1000 * 1000);
 	if (IS_ERR(ap806_clks[2])) {
 		ret = PTR_ERR(ap806_clks[2]);
-- 
1.7.9.5

